extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -223,6 +280,21 @@ export class ProjectsController {
 
 	//#region Helper methods
 
+	public getDeployDialog(project: Project): DeployDatabaseDialog {
+		return new DeployDatabaseDialog(this.apiWrapper, project);
+	}
+
+	private static async getService(extensionId: string): Promise<mssql.IDacFxService> {
+		const ext = extensions.getExtension(extensionId);
+
+		if (ext) {
+			return (ext.exports as mssql.IExtension).dacFx;
```

Comment:
The preferred way is to call ext.activate() and the return value from that is the exported API. That way you're guaranteed that it's been activated.